google/cloud/aiplatform/utils/worker_spec_utils.py

```diff
@@ -152,17 +161,52 @@ def pool_specs(
         spec_order = [
             self.chief_spec,
             self.worker_spec,
-            self.parameter_server_spec,
+            self.server_spec,
             self.evaluator_spec,
         ]
         specs = [s.spec_dict for s in spec_order]
+        last_non_empty_seen = False
         for i in reversed(range(len(spec_order))):
             if spec_order[i].is_empty:
-                specs.pop()
+                if not last_non_empty_seen:
+                    specs.pop()
+                else:
+                    specs[i] = {}
             else:
-                break
+                last_non_empty_seen = True
         return specs
 
+    def reduction_server_pool(
+        self,
+        reduction_server_replica_count: int = 0,
+        reduction_server_machine_type: str = "n1-highcpu-16",
```

Comment:
Does the service allow a value of None for this? If so, prefer to defer the default to the service instead of encoding it in the client SDK.